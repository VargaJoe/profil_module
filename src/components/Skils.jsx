import React, { Component } from 'react';

class Skils extends Component {

	constructor(props) {
	    super(props);
	    this.state = {};
  	}

	render() {
		let slills_list = this.props.skills.split(";").map((skill) =>
			<li className="skils__item" key={skill.toString()}>
				<span className="btn_link">{skill.toString()}</span>
			</li>
		);

		return (
			<ul className="skils">
				{slills_list}
			</ul>
		);
	}
};

export default Skils;
