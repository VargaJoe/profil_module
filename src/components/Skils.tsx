import * as React				from 'react';

class Skils extends React.Component<any, any> {

	render() {
		let slillsList = this.props.skills.split(';').map((skill: string) =>
			(
				<li className="skils__item" key={skill.toString()}>
					<span className="btn_link">{skill.toString()}</span>
				</li>
			)
		);

		return (
			<ul className="skils">
				{slillsList}
			</ul>
		);
	}
}

export default Skils;
