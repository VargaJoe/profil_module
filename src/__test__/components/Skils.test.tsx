import * as React 						  	from 'react';
import Skils 							    from '../../components/Skils';
import { 
	configure, 
		shallow,
		mount } 							from 'enzyme';
import * as Adapter 					  	from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<Skils /> shallow rendering', () => {

    const props = {
        skills: 'item1;item2;item3;item4'
    };
		
	const skills = shallow(<Skils {...props}/>); 
		
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(skills)).toMatchSnapshot();
	});
		
});
 