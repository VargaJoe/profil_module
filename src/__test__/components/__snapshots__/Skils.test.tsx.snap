// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skils /> shallow rendering Match to snapshot 1`] = `
<ul
  className="skils"
>
  <li
    className="skils__item"
    key="item1"
  >
    <span
      className="btn_link"
    >
      item1
    </span>
  </li>
  <li
    className="skils__item"
    key="item2"
  >
    <span
      className="btn_link"
    >
      item2
    </span>
  </li>
  <li
    className="skils__item"
    key="item3"
  >
    <span
      className="btn_link"
    >
      item3
    </span>
  </li>
  <li
    className="skils__item"
    key="item4"
  >
    <span
      className="btn_link"
    >
      item4
    </span>
  </li>
</ul>
`;
